# 카이제곱 검정
: 두 범주형 변수가 서로 상관이 있는지 판단하는 통계적 검정 방법.
예) 학력, 성별, 직업의 만족도, 

### 아동별 장난감 보유 현황
- 귀무가설 : child1과 child2의 장난감 보유수에 대한 만족도는 차이가 없다. 
- 대립가설 : child1과 child2의 장난감 보유수에 대한 만족도는 차이가 있다.

In [3]:
# child별 장난감 보유 현황
child1 <- c(5, 11, 1)
child2 <- c(4, 7, 3)

In [5]:
# 분석을 위해 데이터 합치기
Toy <- cbind(child1, child2)
Toy

child1 child2
[1,]  5     4     
[2,] 11     7     
[3,]  1     3

In [7]:
rownames(Toy) <- c("car","truck","doll")
Toy

child1 child2
car    5     4     
truck 11     7     
doll   1     3

In [9]:
chisq.test(Toy)

Warning message in chisq.test(Toy):
“Chi-squared approximation may be incorrect”


	Pearson's Chi-squared test

data:  Toy
X-squared = 1.7258, df = 2, p-value = 0.4219


### 결론
p-value = 0.4219 > 0.05(95% 신뢰구간) -> 귀무가설 채택
> child1과 child2dml 데이터는 통계적으로 큰 차이가 없다.

p-value는 0에 가까울수록 좋다. 미리 정해진 유의수준(일반적으로 0.05) 보다 작으면 대립가설을 채택하고 유의수준보다 크면 귀무가설을 채택.

In [12]:
cor(child1, child2)

[1] 0.9862414

### 메세지 : 카이제곱 근사값이 정확하지 않을 수도 있다.
- 데이터량이 적은 경우 발생

In [13]:
# 피셔검정

ERROR: Error in eval(expr, envir, enclos): 객체 '피셔검정'를 찾을 수 없습니다


In [15]:
fisher.test(Toy)


	Fisher's Exact Test for Count Data

data:  Toy
p-value = 0.5165
alternative hypothesis: two.sided


#### p-value=0.5165>0.05 -->귀무가설

## t-검정

In [18]:
# 귀무가설 : 건전지 수명은 1000시간이다.
# 대립가설 : 건전지 수명은 1000시간이 아니다.

In [19]:
bat <- c(980, 1000, 868, 1032, 1012, 1002, 996, 1017)

In [20]:
# 데이터의 분포가 정규분포이닞 확인 -> Shapiro-Wilk 검정을 실시
# 귀무가설 : 자료가 정규분포를 따른다.
# 대립가설 : 자료가 정규분포를 따르지 않는다.

[1]  980 1000  868 1032 1012 1002  996 1017

In [21]:
shapiro.test(bat)


	Shapiro-Wilk normality test

data:  bat
W = 0.71139, p-value = 0.003028


In [23]:
# t-검정 실시
t.test(bat, mu=1000, alternative = "two.sided")
#mu 비교하는 대상의 평균
#alternative
# - two.sided : 데이터가 평균과 다르다
# - greater : 데이터가 평균보다 크다
# - less : 데이터가 평균보다 적다.


	One Sample t-test

data:  bat
t = -0.64449, df = 7, p-value = 0.5398
alternative hypothesis: true mean is not equal to 1000
95 percent confidence interval:
  945.7229 1031.0271
sample estimates:
mean of x 
  988.375 


### p-value = 0.5398 > 0.05 --> 귀무가설 : 건전지의 수명은 100시간이다. 